src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/SaveCoreCaseDataService.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.cmc.claimstore.services.ccd;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.ccd.client.StartForCaseworkerApi;
+import uk.gov.hmcts.cmc.ccd.client.SubmitForCaseworkerApi;
+import uk.gov.hmcts.cmc.ccd.client.exception.InvalidCaseDataException;
+import uk.gov.hmcts.cmc.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.cmc.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.cmc.ccd.client.model.Event;
+import uk.gov.hmcts.cmc.ccd.client.model.EventRequestData;
+import uk.gov.hmcts.cmc.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCase;
+
+import java.io.IOException;
+
+@Service
+@ConditionalOnProperty(prefix = "feature_toggles", name = "core_case_data", havingValue = "true")
+public class SaveCoreCaseDataService {
+
+    private StartForCaseworkerApi startForCaseworkerApi;
+    private SubmitForCaseworkerApi submitForCaseworkerApi;
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    public SaveCoreCaseDataService(
+        final StartForCaseworkerApi startForCaseworkerApi,
+        final SubmitForCaseworkerApi submitForCaseworkerApi,
+        final ObjectMapper objectMapper
+    ) {
+        this.startForCaseworkerApi = startForCaseworkerApi;
+        this.submitForCaseworkerApi = submitForCaseworkerApi;
+        this.objectMapper = objectMapper;
+    }
+
+    public CaseDetails save(
+        final String authorisation,
+        final String serviceAuthorisation,
+        final EventRequestData eventRequestData,
+        final CCDCase ccdCase
+    ) {
+
+        JsonNode data = toJson(ccdCase);
```

Comment:
Please move it down, no need to declare fields 19 lines away from the place it is really needed. I do not mind you doing `.data(toJson(ccdCase))`.